aten/src/ATen/cpu/vec/vec256/vec256_float_neon.h

```diff
@@ -466,8 +466,8 @@ template <> class Vectorized<float> {
     return map(std::tanh);
   }
   Vectorized<float> trunc() const {
-    float32x4_t r0 = vcvtq_f32_s32(vcvtq_s32_f32(values.val[0]));
-    float32x4_t r1 = vcvtq_f32_s32(vcvtq_s32_f32(values.val[1]));
+    float32x4_t r0 = vrndq_f32(values.val[0]);
```

Comment:
What is the specific intsance of test that is failing. vcvtq_s32_f32 should round toward zero. vcvtq_f32_s32 does rounding using whatever rounding mode is specified in control register. So maybe thats causing the issue.

Reply:
No, the problem are for floating point values that could not be represented as integers, (for example -1e20 becomes -INT_MAX and becomes -1e9 when converted back), see this [gist](https://gist.github.com/malfet/c612c9f4b3b5681ca1b2a69930825871):
```
inp= 3.1 2.7 -2.9 -1e+20
old_trunc= 3 2 -2 -2.14748e+09
new_trunc= 3 2 -2 -1e+20
```
Besides previous code was two instructions per register where new one is just one, where is harm in that? ;)

Reply:
No not at all. Definitely better. Just wanted to understand what was causing the issue. But yeah what you said makes sense.